integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiSampleWlsMain.java

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2020, 2021, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes;
+
+import java.util.List;
+import java.util.Map;
+
+import oracle.weblogic.kubernetes.annotations.IntegrationTest;
+import oracle.weblogic.kubernetes.annotations.Namespaces;
+import oracle.weblogic.kubernetes.utils.ItMiiSampleHelper;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable;
+
+import static oracle.weblogic.kubernetes.TestConstants.WEBLOGIC_IMAGE_NAME;
+
+/**
+ * Tests to verify MII sample.
+ */
+@DisplayName("Test model in image sample with WLS domain")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@IntegrationTest
+public class ItMiiSampleWlsMain {
+
+  /**
+   * Install Operator.
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *        JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void init(@Namespaces(3) List<String> namespaces) {
+    ItMiiSampleHelper.setDomainType(ItMiiSampleHelper.DomainType.WLS);
+    ItMiiSampleHelper.setImageType(ItMiiSampleHelper.ImageType.MAIN);
+    ItMiiSampleHelper.initAll(namespaces);
+  }
+
+  /**
+   * Generate sample and verify that this matches the source
+   * checked into the mii sample git location.
+   */
+  @Test
+  @Order(1)
+  @DisabledIfEnvironmentVariable(named = "SKIP_CHECK_SAMPLE", matches = "true")
+  @DisplayName("Test to verify MII Sample source")
+  public void testCheckMiiSampleSource() {
+    Map<String, String> envMap = ItMiiSampleHelper.getEnvMap();
+    envMap.remove("BASE_IMAGE_NAME");
+    ItMiiSampleHelper.execTestScriptAndAssertSuccess(
+        ItMiiSampleHelper.DomainType.WLS,
+        "-check-sample",
+        "Sample source doesn't match with the generated source");
+    envMap.put("BASE_IMAGE_NAME", WEBLOGIC_IMAGE_NAME);
+    ItMiiSampleHelper.setEnvMap(envMap);
```

Comment:
minor: This is the only test that has its impl in a subclass, and the only subclass that accesses internals to the helper like env-map and 'execTest'. For consistency and slightly better encapsulation, consider moving it to the helper as 'callCheckMiiSampleSource' and removing the associated getEnvMap/setEnvMap methods...

minor: Consider putting the restore of the image name in a 'finally' since future tests need the correct value for the image name. Technically it doesn't really matter at present, as 'execTest' won't allow any more tests to run once a test fails, but this would make the code a little bit more self documenting in my opinion, plus make it a little more robust if someone ever decides to let a future test run even if this test fails...

Reply:
changed